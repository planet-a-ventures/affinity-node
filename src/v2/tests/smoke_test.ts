import { afterEach, describe, it } from '@std/testing/bdd'
import { apiKey, isLiveRun } from '../../v1/tests/env.ts'

import { assertSnapshot } from '@std/testing/snapshot'
import fetchMock from 'fetch-mock'
import {
    AuthApi,
    CompaniesApi,
    createConfiguration,
    paginated,
} from '../index.ts'

describe('whoami', () => {
    const config = createConfiguration({
        authMethods: {
            bearerAuth: {
                tokenProvider: {
                    getToken: async () => {
                        return await apiKey() || 'api_key'
                    },
                },
            },
        },
    })

    afterEach(() => {
        fetchMock.reset()
    })

    it('can be called', async (t) => {
        if (!isLiveRun()) {
            fetchMock.get('https://api.affinity.co/v2/auth/whoami', {
                'tenant': {
                    'id': 123,
                    'name': 'Planet A',
                    'subdomain': 'planet',
                },
                'user': {
                    'id': 1234567,
                    'firstName': 'Joscha',
                    'lastName': 'Feth',
                    'emailAddress': 'j@p-a.com',
                },
                'grant': {
                    'type': 'api-key',
                    'scopes': ['api'],
                    'createdAt': '2024-06-24T10:22:21Z',
                },
            })
        }
        const authApi = new AuthApi(config)

        const auth = await authApi.getV2AuthWhoami()
        await assertSnapshot(t, auth)
    })

    it('can read a list and page through it', async (t) => {
        if (!isLiveRun()) {
            fetchMock.get('https://api.affinity.co/v2/companies?limit=1', {
                'data': [
                    {
                        'id': 123,
                        'name': 'Planet A',
                        'domain': 'planet-a.com',
                        'domains': ['planet-a.com'],
                        'isGlobal': true,
                        'fields': undefined,
                    },
                ],
                'pagination': {
                    'prevUrl': null,
                    'nextUrl':
                        'https://api.affinity.co/v2/companies?cursor=123',
                },
            })
            fetchMock.get(
                'https://api.affinity.co/v2/companies?cursor=123&limit=1',
                {
                    'data': [
                        {
                            'id': 123,
                            'name': 'Planet B',
                            'domain': 'planet-b.com',
                            'domains': ['planet-b.com'],
                            'isGlobal': false,
                            'fields': undefined,
                        },
                    ],
                    'pagination': {
                        'prevUrl': 'https://api.affinity.co/v2/companies',
                        'nextUrl': null,
                    },
                },
            )
        }
        const companiesApi = new CompaniesApi(config)

        for await (
            const page of paginated(
                companiesApi.getV2Companies.bind(companiesApi),
            )({
                limit: 1,
            })
        ) {
            await assertSnapshot(t, page)
        }
    })
})
