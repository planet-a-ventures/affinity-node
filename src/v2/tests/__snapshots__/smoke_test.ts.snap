export const snapshot = {};

snapshot[`V2 > can be called 1`] = `
WhoAmI {
  grant: Grant {
    createdAt: 2024-06-24T10:22:21.000Z,
    scopes: [
      "api",
    ],
    type: "api-key",
  },
  tenant: Tenant {
    id: 123,
    name: "Planet A",
    subdomain: "planet",
  },
  user: User {
    emailAddress: "j@p-a.com",
    firstName: "Joscha",
    id: 1234567,
    lastName: "Feth",
  },
}
`;

snapshot[`V2 > can read a list and page through it 1`] = `
[
  Company {
    domain: "planet-a.com",
    domains: [
      "planet-a.com",
    ],
    fields: undefined,
    id: 123,
    isGlobal: true,
    name: "Planet A",
  },
]
`;

snapshot[`V2 > can read a list and page through it 2`] = `
[
  Company {
    domain: "planet-b.com",
    domains: [
      "planet-b.com",
    ],
    fields: undefined,
    id: 123,
    isGlobal: false,
    name: "Planet B",
  },
]
`;
